libs/langchain/langchain/schema/runnable/retry.py

```diff
@@ -0,0 +1,250 @@
+from typing import Any, Dict, List, Optional, Tuple, Type, TypeVar, Union, cast
+
+from tenacity import (
+    AsyncRetrying,
+    RetryCallState,
+    RetryError,
+    Retrying,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_exponential_jitter,
+)
+
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForChainRun,
+    CallbackManagerForChainRun,
+)
+from langchain.schema.runnable.base import Input, Output, RunnableBinding
+from langchain.schema.runnable.config import RunnableConfig, patch_config
+
+T = TypeVar("T", CallbackManagerForChainRun, AsyncCallbackManagerForChainRun)
+U = TypeVar("U")
+
+
+class RunnableRetry(RunnableBinding[Input, Output]):
+    """Retry a Runnable if it fails."""
+
+    retry_if_exception_type: Tuple[Type[BaseException]] = (Exception,)
+
+    wait_exponential_jitter: bool = True
+
+    stop_after_attempt: int = 3
```

Comment:
fwiw this param is called `max_attempt_number` in tenacity (which imo is bit clearer)